google/cloud/testing_util/is_proto_equal.cc

```diff
@@ -31,6 +31,38 @@ absl::optional<std::string> CompareProtos(
   return delta;
 }
 
+absl::optional<std::string> CompareProtosApproximately(
+    google::protobuf::Message const& arg,
+    google::protobuf::Message const& value) {
+  std::string delta;
+  google::protobuf::util::DefaultFieldComparator comparator;
+  comparator.set_float_comparison(
+      google::protobuf::util::DefaultFieldComparator::APPROXIMATE);
+  // Keep the comparator's default fraction and margin.
+  google::protobuf::util::MessageDifferencer differencer;
+  differencer.set_field_comparator(&comparator);
+  differencer.ReportDifferencesToString(&delta);
+  auto const result = differencer.Compare(arg, value);
+  if (result) return {};
```

Comment:
I thought we wanted to return `absl::nullopt` in these cases?

Reply:
Done.  Thanks.